File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    cor = {1:0, 2:0, 3:0}
    p1 = [1,2,3,4,5]
    p2 = [2,1,2,3,2,4,2,5]
    p3 = [3,3,1,1,2,2,4,4,5,5]

    for i in range(len(answers)):
        if p1[i % len(p1)] == answers[i]:
            cor[1] += 1
        if p2[i % len(p2)] == answers[i]:
            cor[2] += 1
        if p3[i % len(p3)] == answers[i]:
            cor[3] += 1
    for i in cor.keys():
        if cor[i] == max(cor.values()):
            answer.append(i)
    return answer

File: 프로그래머스/lv1/42840. 모의고사/모의고사.js
function solution(answers) {
    const stu1 = [1,2,3,4,5];
    const stu2 = [2,1,2,3,2,4,2,5];
    const stu3 = [3,3,1,1,2,2,4,4,5,5];
    const answer = [];
    let stu1_a = 0, stu2_a = 0, stu3_a = 0;
    
    for(let i=0; i < answers.length; i++){
        if(answers[i] === stu1[i % stu1.length]) stu1_a+=1;
        if(answers[i] === stu2[i % stu2.length]) stu2_a+=1;
        if(answers[i] === stu3[i % stu3.length]) stu3_a+=1;
    };
    
    const max = Math.max(stu1_a, stu2_a, stu3_a);
    if(stu1_a === max) answer.push(1);
    if(stu2_a === max) answer.push(2);
    if(stu3_a === max) answer.push(3);
    
    return answer;
}